frame/cosmwasm/src/runtimes/wasmi.rs

```diff
@@ -312,18 +314,31 @@ impl<'a, T: Config> VMBase for CosmwasmVM<'a, T> {
 		&mut self,
 		message_hash: &[u8],
 		signature: &[u8],
-		_recovery_param: u8,
+		recovery_param: u8,
 	) -> Result<Result<Vec<u8>, ()>, Self::Error> {
-		let signature: [u8; 65] = match signature.try_into() {
-			Ok(signature) => signature,
-			Err(_) => return Ok(Err(())),
-		};
+		// `recovery_param` must be 0 or 1. Other values are not supported from CosmWasm.
+		if recovery_param > 2 {
+			return Ok(Err(()))
+		}
 
-		let message_hash: [u8; 32] = match message_hash.try_into() {
+		if signature.len() != ECDSA_SIGNATURE_LEN - 1 {
+			return Ok(Err(()))
+		}
+
+		// Try into a [u8; 32]
+		let message_hash = match message_hash.try_into() {
 			Ok(message_hash) => message_hash,
 			Err(_) => return Ok(Err(())),
 		};
 
+		let signature = {
+			// Since we fill `signature_inner` with `recovery_param`, when 64 bytes are written
+			// the final byte will be the `recovery_param`.
+			let mut signature_inner = [recovery_param; ECDSA_SIGNATURE_LEN];
+			signature_inner[..ECDSA_SIGNATURE_LEN - 1].copy_from_slice(signature);
```

Comment:
so this is SUBSTRATE_ECDSA_SIGNATURE_LEN

Reply:
Yes, it would be better to rename to that.